js/components/utils.js

```diff
@@ -431,34 +431,38 @@ var Utils = {
    * @function arrayDeepMerge
    * @param {Array} target - An array that will receive new items if additional items are passed
    * @param {Array} source - An array containing additional items to merge into target
-   * @param {Object} optionsArgument - optional parameters passed, i.e. arrayMerge function
+   * @param {Object} optionsArgument - optional parameters passed, i.e. arrayMerge, swap, unionBy, clone
    * @returns {Array} new merged array with items from both target and source
    */
   arrayDeepMerge: function(target, source, optionsArgument) {
+    var targetArray = optionsArgument.swap ? source : target;
```

Comment:
also this `arrayDeepMerge` function is only used to merge an array of objects. actual objects in the main deep merge still need to maintain target/source hierarchy. this function changes how the array list of objects are looped to maintain correct sort order while maintaining target/source hierarchy of original merge.

will retest while working on unit test